File: fizzbuzz.ml
let fizzbuzz n =
  match n mod 3, n mod 5 with
  | 0, 0 -> "FizzBuzz"
  | 0, _ -> "Fizz"
  | _, 0 -> "Buzz"
  | _ -> string_of_int n

let () =
  for i = 1 to 100 do
    Printf.printf "%s\n" (fizzbuzz i)
  done

File: fizzbuzz.erl
-module(fizzbuzz).
-export([main/0]).

fizzbuzz(N) when N rem 15 =:= 0 -> "FizzBuzz";
fizzbuzz(N) when N rem 3 =:= 0 -> "Fizz";
fizzbuzz(N) when N rem 5 =:= 0 -> "Buzz";
fizzbuzz(N) -> integer_to_list(N).

main() ->
    lists:foreach(fun(N) -> io:format("~s~n", [fizzbuzz(N)]) end, lists:seq(1, 100)).
